packages/node_modules/node-red/red.js

```diff
@@ -142,208 +142,278 @@ if (process.env.NODE_RED_ENABLE_PROJECTS) {
     settings.editorTheme.projects.enabled = !/^false$/i.test(process.env.NODE_RED_ENABLE_PROJECTS);
 }
 
+var httpsPromise;
 if (settings.https) {
-    server = https.createServer(settings.https,function(req,res) {app(req,res);});
-} else {
-    server = http.createServer(function(req,res) {app(req,res);});
-}
-server.setMaxListeners(0);
+    var startupHttps = settings.https;
 
-function formatRoot(root) {
-    if (root[0] != "/") {
-        root = "/" + root;
-    }
-    if (root.slice(-1) != "/") {
-        root = root + "/";
+    if (typeof startupHttps === "function") {
+        // Get the result of the function, because createServer doesn't accept functions as input
+        startupHttps = startupHttps(); 
+    }  
+
+    if (startupHttps.hasOwnProperty('then') && typeof startupHttps.then === 'function') {
+        // A promise was returned
+        httpsPromise = startupHttps;
+    } else {
+        // An object was returned - wrap in a promise
+        httpsPromise = Promise.resolve(startupHttps);
     }
```

Comment:
I have further simplified this:
```
var httpsPromise;
if (settings.https) {
    var startupHttps = settings.https;

    if (typeof startupHttps === "function") {
        // Get the result of the function, because createServer doesn't accept functions as input
        startupHttps = startupHttps(); 
    }  
    httpsPromise = Promise.resolve(startupHttps);
}
else {
    // No https is enable - wrap null
    httpsPromise = Promise.resolve(null);
}
```
To this:
```
var startupHttps = settings.https;
if (typeof startupHttps === "function") {
    // Get the result of the function, because createServer doesn't accept functions as input
    startupHttps = startupHttps(); 
}  
var httpsPromise = Promise.resolve(startupHttps);
```
Since - at the start of the 'then' part - it was checked anyway whether the startupHttps is defined:
```
httpsPromise.then(function(startupHttps) {
    if (startupHttps) {
      ...
   }
}
```
Still seems to be working without https settings...